src/plugins/dataflow/nodes/controls/dropdown-list-control.tsx

```diff
@@ -79,21 +83,18 @@ export class DropdownListControl extends Rete.Control {
                                     });
       const option = options.find((opt) => optionValue(opt) === val);
       const name = option?.name ?? val;
+      const displayName = option?.displayName ?? name;
       const icon = option?.icon?.({}) || null;
       const activeHub = (option as any).active;
       const liveNode = this.getNode().name.substring(0,4) === "Live";
       const disableSelected = this.key === "hubSelect" && liveNode && !activeHub;
       const labelClasses = disableSelected ? "disabled item top" : "item top";
-
+      console.log("| displayName: ", displayName);
       return (
         <div className={`node-select ${listClass}`} ref={divRef}>
           <div className={labelClasses} onMouseDown={handleChange(onItemClick)}>
-            { icon &&
-            <svg className="icon top">
-              {icon}
-            </svg>
-            }
-            <div className="label">{name}</div>
+            { icon && <svg className="icon top">{icon}</svg> }
+            <div className={optionLabelClass(displayName as string)}>{displayName}</div>
```

Comment:
I struggled with the compiler on this - the way I have it written I have to either cast it when I set the variable or cast it here where I pass it.  I think it's because displayName could be undefined, because one path sets it to `option?.displayName`

Reply:
In the scenario where `displayName` is `undefined`, this will result in adding a class of `undefined`. Rather than casting it away you should deal with the possibility of an `undefined` value, e.g. by using `displayName ?? ""`.

Reply:
@kswenson ah, ok, thanks will add this to upcoming additional work on this

Reply:
I ended up doing this: https://github.com/concord-consortium/collaborative-learning/pull/1781/commits/3b778980c5464de49e418da9d3e9dfbc549fc3ce

It was because things are already setup so that if there is no "name" val is used for the name, but val _could_ be a number.   So in the case that we want it to be the basis for a class, we cast it as a string before setting `name`.   I believe the only case where this would apply is when we choose a number from the generator drop downs.